kore/src/Kore/Step/Remainder.hs

```diff
@@ -153,3 +167,50 @@ substitutionConditions subst =
   where
     substitutionCoverageWorker (x, t) =
         Syntax.Predicate.makeEqualsPredicate (mkVar x) t
+
+introduceDefinedness
+    :: forall variable
+    .  ( SortedVariable variable
+       , FreshVariable variable
+       , Unparse variable
+       , Show variable
+       )
+    => OrPredicate variable
+    -> Pattern variable
+    -> Pattern variable
+introduceDefinedness cond result =
+    foldr (flip andCondition) result cond
```

Comment:
This seems strange. It seems that you have the definedness condition as `P1 or P2 or ...` and the result `R` and you're transforming it into `R and P1 and P2 and ...`. Why does that make sense? If it makes sense, could you add a comment explaining it?

From the function signature, I would have expected something like `(R and P1) or (R and P2) or ...`, but I'm not sure what you're trying to do.

Is this a generic function that should go to `OrPredicate.hs` (or `OrPattern.hs` or somehting similar), or is it something specific to definedness?

Reply:
> This seems strange. It seems that you have the definedness condition as P1 or P2 or ... and the result R and you're transforming it into R and P1 and P2 and .... Why does that make sense? If it makes sense, could you add a comment explaining it?

The `ceil` of the child of the function application is an `OrPredicate`, and that function is intended to be used when mapping over the remainders, so this will result in `(remainder1 and ceilChild) or (remainder2 and ceilChild) ...`. I think the error is that `ceilChild` should be folded to a disjunction not a conjunction (since it's an `OrPredicate`). Would this be correct?